### ISSUE TITLE
Address validation should take multipolygons into account

### ISSUE THREAD

**Issue by MinerMax555** [osm-fr/osmose-backend#794]
Item 2060
Class 19

Hey,
so while resolving issues in a city near me I discovered that some issues persist near what is pretty much the main square of the city, even after I changed the addr:street tags of the addresses around to match those of the square. I suspect this has to do with the fact that this square is mapped as a multipolygon.
Have a look at [this](http://osmose.openstreetmap.fr/de/error/a5dc122d-6ce9-95ac-fcbc-cdfbdc18c709) issue and it's neighbors.
Square this address belongs to: [https://www.openstreetmap.org/relation/4003691](https://www.openstreetmap.org/relation/4003691)

would be really nice if you could either fix this behaviour or tell me if I am overlooking something. 
Thanks in advance

**Comment by Bibi56** [comment]
If it's a square, it should be described with place=square and then the addr field to use is addr:place, not adddr:street

**Comment by frodrigo** [comment]
The relation have not `place` tag.

Yes. It is linked to the missing relation support in this analysis.

`highway` is not build using relation 
https://github.com/osm-fr/osmose-backend/blob/master/analysers/Analyser_Osmosis.py#L36

**Comment by Bibi56** [comment]
Note: each address, like https://www.openstreetmap.org/node/2749571952 should also use `addr:place` instead of `addr:street`. Yes for multilingual places it would be good `associatedStreet` would also support `addr:place` to reduce duplication.

**Comment by frodrigo** [comment]
@Bibi56 non, no `addr:place` here since the square (plazza) is not  a `place=*` but rather a `highway=pedestrian`.

**Comment by frodrigo** [comment]
@MinerMax555 Did you mark the issues as false-positives?

**Comment by frodrigo** [comment]
https://github.com/osm-fr/osmose-backend/blob/master/analysers/analyser_osmosis_relation_associatedStreet.py#L570

**Comment by Bibi56** [comment]
> @Bibi56 non, no `addr:place` here since the square (plazza) is not a `place=*` but rather a `highway=pedestrian`.

But as I said, a plazza _should_ be tagged place=square, as you mention it's a square,
